### DISCUSSION TITLE
retrieving text from a .ldb-cache-file created by SEB?

### DISCUSSION THREAD

**Discussion by ElArt0** [SafeExamBrowser/seb-win-refactoring#561] [Q&A]
Hello
Teacher here that used the Safe Exam Browser in combination with a online exam tool (istest2.ch). 

The online exam tool uses the SEB-Cache until a student manually safes a question in the exam tool - with essays that typically happens in the end, once they submit the essay... Only then is an answer to a test question saved on the server of the exam tool and can be retrieved from there by the developers. 

A crash (or rather a session timeout) happened when she wanted to submit her essay after 4 hours of writing. Normally that's not a problem as the exam tool retrieves the data from the SEB-cache once a student logs in another time. The online exam tool hadn't updated the caching path the new Safe Exam Browser 3.4.1 uses though, so after the session timeout, a student lost 4 hours of essay writing. The essay was still in the cache, but the exam tool didn't find it.

I was able to find the latest version of her esssay in her local SEB-cache, but it's a .ldb-file and I don't know how to retrieve the data from it. I clearly see that it's the essay right there and parts are readable, but I cannot reconstruct the essay. Does anybody know how the file can be openend to retrieve the essay? (see file in the attachment - because github doesn't support *.ldb-Files I renamed it as .txt - but it's an .ldb-File.
[000023.txt](https://github.com/SafeExamBrowser/seb-win-refactoring/files/10551359/000023.txt)

**Comment by dbuechel** [comment]
There are various options which might clear the SEB cache when restarting, see https://safeexambrowser.org/windows/win_usermanual_en.html#BrowserPane > Browser Security as well as https://safeexambrowser.org/windows/win_usermanual_en.html#ExamPane > Session Handling.

> A crash (or rather a session timeout) happened when she wanted to submit her essay after 4 hours of writing.

It would probably make a difference whether SEB crashed or whether the browser session timed out and the online exam tool failed to subsequently extract the essay from the SEB browser cache.

> The online exam tool hadn't updated the caching path the new Safe Exam Browser 3.4.1 uses though, so after the session timeout, a student lost 4 hours of essay writing.

This statement isn't quite clear, could you please elaborate: The exam tool cannot influence the file path SEB uses for its browser cache. However, it can obviously select the caching strategy / technology it uses (e.g. Indexed DB).

As far as I can determine from afar, this rather looks like an issue with the online examination tool itself. Did you try to reach out to its developers? They should be able to extract the data you need from the file you obtained from the student's system.

**Reply by ElArt0** [reply]
It wasn't the SEB that crashed, but the online exam tool that had a session timeout and couldn't retrieve the data from the cache, so it's definitely a problem with the online exam tool as they hadn't updated on the new SEB 3.4.1 cache path as one of the developers had to admit. They only found out through my feedback that the caching path had changed. The exam tool tried to retrieve the cache, but as it pointed to a different (the old) path, it didn't find anything. 
I asked the developers what they could do, but they can't do anything about it in the short term, hence I was asking here. I'm so close to retrieving the data as I got the right cache file, yet I can't read it (or only parts).

**Reply by dbuechel** [reply]
We most unfortunately can't really help you much further here, as this is functionality which resides completely within the browser engine. You might have some luck posting your question in the CEF forum, see https://magpcss.org/ceforum/.

**Reply by ElArt0** [reply]
Thanks for the referral. I'll check there.
